### PR TITLE
Refactor: get rid of reflect package

### PR DESCRIPTION
**Pull request by alexandear**
This PR replaces `reflect` package in `device/device.go` with more simple language constructs.

### CODE REVIEW

File: device/device.go [hunk 1]
```diff
@@ -52,7 +51,7 @@ func Open(path string, options ...Option) (*Device, error) {
 	dev.cap = cap
 
 	// set preferred device buffer size
-	if reflect.ValueOf(dev.config.bufSize).IsZero() {
```
Comment (vladimirvivien) [inline device/device.go:55]: Hi @alexandear the reflect package was used here to detect whether a value has never been set, hence test for its zero-value, or if it was set to the value zero.  But, looking back at the code, I think these tested values cannot have valid V4L2 zero values anyway, so this is OK.

File: device/device.go [hunk 2]
```diff
@@ -200,7 +199,7 @@ func (d *Device) GetPixFormat() (v4l2.PixFormat, error) {
 		return v4l2.PixFormat{}, v4l2.ErrorUnsupportedFeature
 	}
 
-	if reflect.ValueOf(d.config.pixFormat).IsZero() {
+	if d.config.pixFormat.IsZero() {
```
Comment (vladimirvivien) [inline device/device.go:202]: I am not convinced we need the additional method on type PixFormat, just to test for its zero-ness. Since this is only done here (in the entire code base), just do the test here:

```go
if (d.config.pixFormat == PixFormat{}) { ... } // i think this will work
```

File: v4l2/format.go [hunk 3]
```diff
@@ -278,6 +278,10 @@ func (f PixFormat) String() string {
 	)
 }
 
+func (f PixFormat) IsZero() bool {
```
Comment (vladimirvivien) [inline v4l2/format.go:281]: See comment above.

**Review by vladimirvivien — COMMENTED** [review]
Thanks @alexandear the contribution. This looks good, I left a couple of comments.

File: device/device.go (hunk 2 above)
Comment (alexandear) [reply device/device.go]: This works:
`if d.config.pixFormat == (PixFormat{}) { ... } `

We need to add `()` around `PixFormat{}` to satisfy the Go compiler.

**Review by vladimirvivien — APPROVED** [review]
Looks good. Thanks!
